_bleio/common.py

```diff
@@ -327,15 +328,15 @@ def _start_scan_hcitool(
             )
         # pylint: enable=consider-using-with
         # Throw away the first two output lines of hcidump because they are version info.
-        hcidump.stdout.readline()
-        hcidump.stdout.readline()
+        hcidump.stdout.readline()  # type: ignore[union-attr]
+        hcidump.stdout.readline()  # type: ignore[union-attr]
         returncode = self._hcitool.poll()
         start_time = time.monotonic()
-        buffered = []
+        buffered: List[bytes] = []
         while returncode is None and (
-            not timeout or time.monotonic() - start_time < timeout
+            timeout is None or time.monotonic() - start_time < timeout
         ):
-            line = hcidump.stdout.readline()
+            line = hcidump.stdout.readline()  # type: ignore[union-attr]
```

Comment:
I think I lean toward leaving out the `# type: ignore` comments for now since we don't have mypy in the CI actions yet. 

When we do add it we'll decide as a group what configuration we want to use. Hopefully there will be a single config file that can get added to all repos like we do with pre-commit and pylint and avoid the need to disable things individually like this as much as possible. 

That being said, it's worth talking about it with the rest of the team. I'll add a note to discuss it in the weeds during next weeks meeting. If you're around we'd love to have you attend on discord if you'd like to be involved in the discussion.

Reply:
I will of course do whatever the core devs think is best, and I'm happy to join the next meeting if I can!

That said, to explain my thinking here - this is a static type error because `Popen` initially sets `self.stdout` & friends to None. But given that you specify `stdout=subprocess.PIPE`, [we can rely on `stdout` being defined](https://docs.python.org/3/library/subprocess.html?highlight=popen%20run#subprocess.Popen.stdout). So, I would NOT advocate for a blanket ignore of union-attr (or perhaps some more specific directive) - just in this case where we know that dynamically we will definitely have a defined stdout.

All that said, I'm grokking that the mission here is to make the code more intelligible to users, NOT necessarily getting mypy to be quiet.

Reply:
> All that said, I'm grokking that the mission here is to make the code more intelligible to users, NOT necessarily getting mypy to be quiet.

I would definitely agree with this sentiment. We want to make the code better where we can, but understand-ability for beginner users is a very large focus. In cases where mypy errors run counter to that I would lean toward ignoring or quieting the error and going with the code that is easiest for folks (potentially without much programming experience) to understand.